### PR TITLE
[bugfix] Fix bug in unloading of conflicting modules

### LINKED ISSUE #59: Unnecessary 'module unload' commands in the generated job script
Here is an example output for the `PrgEnv-pgi` environment:

```bash
#!/bin/bash -l
#SBATCH --job-name="rrtmgp_check_daint_gpu_PrgEnv-pgi"
#SBATCH --time=0:10:0
#SBATCH --ntasks=1
#SBATCH --partition=debug
#SBATCH --exclusive
#SBATCH --output=/scratch/snx3000/karakasv/reframe-tests/stage/gpu/rrtmgp_check/PrgEnv-pgi/rrtmgp_check.out
#SBATCH --error=/scratch/snx3000/karakasv/reframe-tests/stage/gpu/rrtmgp_check/PrgEnv-pgi/rrtmgp_check.err
#SBATCH --constraint=gpu
module load daint-gpu
module unload PrgEnv-cray
module load PrgEnv-pgi
module unload PrgEnv-cray
module load craype-accel-nvidia60
module unload PrgEnv-cray
module load cray-netcdf
module unload PrgEnv-cray
module unload pgi
module load pgi/17.7.0
```

I think the problem comes from the way we record the conflicts.

Internal issue: https://madra.cscs.ch/scs/reframe/issues/565

### PR DESCRIPTION
**Pull request by teojgo**
* Fix bug which led to accumulation of module unloading.

* Use raw string to avoid warnings.

Fixes #59

### CODE REVIEW

**Comment by vkarak** [comment]
@teojgo Good, I get it now.

File: reframe/core/environments.py [hunk 1]
```diff
@@ -79,8 +79,9 @@ def load(self):
             if rt.modules_system.is_module_loaded(m):
                 self._preloaded.add(m)
 
-            self._conflicted += rt.modules_system.load_module(m, force=True)
-            for conflict in self._conflicted:
+            module_conflicts = rt.modules_system.load_module(m, force=True)
+            self._conflicted += module_conflicts
+            for conflict in module_conflicts:
```
Comment (vkarak) [inline reframe/core/environments.py]: The loop variable should be simply `m`.

File: reframe/core/environments.py [hunk 2]
```diff
@@ -79,8 +79,9 @@ def load(self):
             if rt.modules_system.is_module_loaded(m):
                 self._preloaded.add(m)
 
-            self._conflicted += rt.modules_system.load_module(m, force=True)
-            for conflict in self._conflicted:
+            module_conflicts = rt.modules_system.load_module(m, force=True)
+            self._conflicted += module_conflicts
```
Comment (vkarak) [inline reframe/core/environments.py]: Move this assignment after the loop. Placed here, it kind of lets you expect that `self._conflicted` will be used just right after.

File: reframe/core/environments.py [hunk 3]
```diff
@@ -79,8 +79,9 @@ def load(self):
             if rt.modules_system.is_module_loaded(m):
                 self._preloaded.add(m)
 
-            self._conflicted += rt.modules_system.load_module(m, force=True)
-            for conflict in self._conflicted:
+            module_conflicts = rt.modules_system.load_module(m, force=True)
```
Comment (vkarak) [inline reframe/core/environments.py]: I'd name this simply `conflicted`.

File: reframe/core/environments.py (hunk 1 above)
Comment (teojgo) [reply reframe/core/environments.py]: its a nested loop and the outer one already uses `m`

**Pull request merged**